minidump-stackwalk/stackwalker.cc

```diff
@@ -1411,9 +1430,22 @@ int main(int argc, char** argv)
   StackFrameSymbolizerForward symbolizer(symbol_supplier.get(), &resolver);
   MinidumpProcessor minidump_processor(&symbolizer, true);
   ProcessState process_state;
+
   ProcessResult result =
     minidump_processor.Process(&minidump, &process_state);
 
+  int arch_bits = 64;
+  if (result == google_breakpad::PROCESS_OK) {
+    if (process_state.system_info()->cpu == "x86") {
```

Comment:
It's better to call [minidump->GetSystemInfo() here](https://searchfox.org/mozilla-central/rev/659198ec55f5a184bee85759a4b3bf9ea0a4e506/toolkit/crashreporter/google-breakpad/src/google_breakpad/processor/minidump.h#1262). Then you can get the raw system info with [SystemInfo::system_info()](https://searchfox.org/mozilla-central/rev/659198ec55f5a184bee85759a4b3bf9ea0a4e506/toolkit/crashreporter/google-breakpad/src/google_breakpad/processor/minidump.h#724) which contains the [processor_architecture field](https://searchfox.org/mozilla-central/rev/659198ec55f5a184bee85759a4b3bf9ea0a4e506/toolkit/crashreporter/google-breakpad/src/google_breakpad/common/minidump_format.h#616). You can match this field against the [list of known architectures](https://searchfox.org/mozilla-central/rev/659198ec55f5a184bee85759a4b3bf9ea0a4e506/toolkit/crashreporter/google-breakpad/src/google_breakpad/common/minidump_format.h#646-668). The only 32-bit ones we care about are `MD_CPU_ARCHITECTURE_X86`, `MD_CPU_ARCHITECTURE_ARM` and `MD_CPU_ARCHITECTURE_X86_WIN64`. Assume that all the others are 64-bit (even though they aren't but we won't encounter them in the wild so we don't care).

I discovered something funky. It's possible to tell breakpad the width of the hex fields, [see this](https://searchfox.org/mozilla-central/rev/659198ec55f5a184bee85759a4b3bf9ea0a4e506/toolkit/crashreporter/google-breakpad/src/processor/minidump.cc#5363) but you have to do it before reading the minidump which kind-of defeats the purpose.

Reply:
Mmm... The `TelemetryEnvironment` has the arch in it. That's pretty deep in the extra information, but only for Firefox--Focus and Fenix don't have that information, so it's not always there.

It's probably easier to look at system info and leave it at that. I'll fix that today.